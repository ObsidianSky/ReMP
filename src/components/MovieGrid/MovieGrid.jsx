import React, { Component } from 'react';
import PropTypes from 'prop-types';
import { connect } from 'react-redux';

import './MovieGrid.scss'
import Movie from './Movie/Movie';

class MovieGrid extends Component {
    getMovies() {
        return this.props.movies.map((movie, index) => (
            <div className="movie-grid__item" key={index}>
                <Movie {...movie}/>
            </div>
        ));
    }

    render() {
        return (
            <div className="movie-grid">
                {this.props.movies.length > 0
                    ? this.getMovies()
                    : <div className="movie-grid__empty">No movies here</div>
                }
            </div>
        )
    }
}

MovieGrid.propTypes = {
    movies: PropTypes.arrayOf(
        PropTypes.shape({
            img: PropTypes.string,
            title: PropTypes.string,
            year: PropTypes.string,
            genre: PropTypes.string
        })
    )
};

const mapStateToProps = state => ({ movies: state.movies });

export default connect(mapStateToProps)(MovieGrid);