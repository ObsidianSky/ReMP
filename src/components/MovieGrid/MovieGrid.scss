@import '../../assets/styles/variables';

.movie-grid {
  padding: 40px 0;
  max-width: 1200px;
  margin: 0 auto;
  display: flex;
  justify-content: flex-start;
  flex-wrap: wrap;
  flex: 1;

  &__item {
    width: 30%;
    margin: 20px 0;

    &:nth-child(3n-1) {
      margin-left: 5%;
      margin-right: 5%;
    }
  }

  &__empty {
    text-transform: uppercase;
    font-size: 26px;
    font-weight: 600;
    color: $text;
    width: 100%;
    text-align: center;
    align-self: center;
  }
}