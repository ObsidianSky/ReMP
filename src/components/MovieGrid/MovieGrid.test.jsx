import React from 'react';
import { shallow } from 'enzyme';
import toJson from 'enzyme-to-json';

import { MovieGrid } from './MovieGrid';

describe('MovieGrid', () => {
    let props;
    let showMovieDetails;

    beforeEach(() => {
        showMovieDetails = jest.fn();

        props = {
            movies: [
                {
                    id: 'id',
                    img: 'img',
                    title: 'title',
                    year: 'year',
                    genre: 'genre',
                    rating: 'rating',
                },
                {
                    id: 'id1',
                    img: 'img1',
                    title: 'title1',
                    year: 'year1',
                    genre: 'genre1',
                    rating: 'rating1',
                }
            ],
            showMovieDetails
        }
    });

    describe('when we have movies', () => {

        test('should renders correctly', () => {
            const wrapper = shallow(<MovieGrid { ...props } />);

            expect(toJson(wrapper)).toMatchSnapshot();
        });

    });

    describe('when we do NOT have movies', () => {
        beforeEach(() => {
            props.movies = [];
        });

        test('should renders correctly', () => {
            const wrapper = shallow(<MovieGrid { ...props } />);

            expect(toJson(wrapper)).toMatchSnapshot();
        });
    });

    describe('when i dont know how to name it at all', () => {
        test('should ', () => {
            const wrapper = shallow(<MovieGrid { ...props } />);
            wrapper.find('Movie').first().simulate('movieClick');

            expect(showMovieDetails).toHaveBeenCalledWith(props.movies[0].id);
        });
    });
});

