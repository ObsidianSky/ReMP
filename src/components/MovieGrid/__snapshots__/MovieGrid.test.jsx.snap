// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`MovieGrid when we do NOT have movies should renders correctly 1`] = `
<div
  className="movie-grid"
>
  <div
    className="movie-grid__empty"
  >
    No movies here
  </div>
</div>
`;

exports[`MovieGrid when we have movies should renders correctly 1`] = `
<div
  className="movie-grid"
>
  <div
    className="movie-grid__item"
    key="0"
  >
    <Movie
      genre="genre"
      id="id"
      img="img"
      onMovieClick={[Function]}
      rating="rating"
      title="title"
      year="year"
    />
  </div>
  <div
    className="movie-grid__item"
    key="1"
  >
    <Movie
      genre="genre1"
      id="id1"
      img="img1"
      onMovieClick={[Function]}
      rating="rating1"
      title="title1"
      year="year1"
    />
  </div>
</div>
`;
